### PR TITLE
create an api design proposal page

### PR DESCRIPTION
**Pull request by gbinal**
I'd propose merging this pull request for convenience and then using further PRs and edits to collaborate on this proposal.

### CODE REVIEW

**Comment by gbinal** [comment]
@rdgsa, @nvembar - When you get a second, game to take a look at this and either merge so that this section could be a sandbox for engaging on this topic or holler with any thoughts?

**Comment by rdgsa** [comment]
@nvembar, @jcastle I like Gray's recommendation to create a standalone proposal page to discuss this in the open and get feedback from the working group. If no objections, I will merge this into the repo and begin putting in issues and pull requests for the discussion around it.

File: api-design/overview.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+| https://api.gsa.gov/documents?id=32330  | Returns a specific document  |
+
+
+##### Proposed next steps 
+
+1. Merge the PR that creates this page.  
+2. Circulate it and more context to the agency API working group.  
+3. Gain consensus on the idea.  
+4. Have a team (perhaps team CTO) agree to chair the effort.  
+5. Update the above with the candidates for the first round of existing APIs that would be incorporated.
```
Comment (rdgsa) [inline api-design/overview.md]: Gray is there a sandbox environment available for api.gsa.gov? This might be helpful in this step.

Comment (gbinal) [inline api-design/overview.md]: There's no sandbox but the one thing about APIs is that if you don't document them, it's pretty hard for people to notice what you're doing.  

Basically, I'd suggest we take a couple of these and set them up in production on api.data.gov and see what we think.  It only takes a few minutes and doesn't affect the underly API so there's not really any risk.

**Comment by gbinal** [comment]
Barring any concerns, okay to merge this and have it serve as a basis for discussion?

**Pull request merged**